execution-engine/src/native_module_manager.rs

```diff
@@ -0,0 +1,293 @@
+    /// limited permissions.
+    /// TODO: use it
+    /*fn teardown_fs(&self) -> FileSystemResult<()> {
+        remove_dir_all(self.native_module_directory.as_path())?;
+        Ok(())
+    }*/
+
+    /// Run the native module. The input is passed by the WASM program via the
+    /// native module's special file.
+    pub fn execute(&mut self, input: Vec<u8>) -> FileSystemResult<()> {
+        if self.native_module.is_static() {
+            // Look up native module in the static native modules table
+            let mut nm = STATIC_NATIVE_MODULES
+                .lock()
+                .map_err(|_| ErrNo::Inval)?;
+            let nm = nm
+                .get_mut(&self.native_module.name().to_string())
+                .ok_or(ErrNo::Inval)?;
+            if nm.try_parse(&input)? {
+                nm.serve(&mut self.native_module_vfs, &input)?;
+            }
+        } else {
+            info!("Preparing the native module's filesystem...");
+            let top_level_files = self.prepare_fs()?;
+            info!("OK");
+
+            // Inject execution configuration into the native module's directory
+            let mut file = File::create(self.native_module_directory.join(EXECUTION_CONFIGURATION_FILE))?;
+            file.write_all(&input)?;
+
+            // Enable SIGCHLD handling in order to synchronously execute the
+            // sandboxer.
+            // This is necessary as Veracruz-Server (Linux) disables SIGCHLD
+            // handling, which is inherited by the runtime manager
+            #[cfg(feature = "std")]
+            unsafe {
+                signal::sigaction(
+                    signal::Signal::SIGCHLD,
+                    &signal::SigAction::new(
+                        signal::SigHandler::SigDfl,
+                        signal::SaFlags::empty(),
+                        signal::SigSet::empty(),
+                    ),
+                )
+                .expect("sigaction failed");
+            }
+
+            info!("Calling sandboxer...");
+            let mount_mappings = self.build_mappings(top_level_files)?;
+            Command::new(NATIVE_MODULE_MANAGER_SANDBOXER_PATH)
+                .args([
+                    "--sandbox2tool_resolve_and_add_libraries",
+                    "--sandbox2tool_mount_tmp",
+                    "--sandbox2tool_additional_bind_mounts",
+                    &mount_mappings,
+                    &self.native_module.entry_point_path().to_str().ok_or(ErrNo::Inval)?.to_owned(),
+                ])
+                .output()?;
+
+            info!("Propagating side effects to the VFS...");
+            self.copy_fs_to_vfs(&PathBuf::from(""))?;
+            info!("OK");
+
+            // TODO:
+            //self.teardown_fs()?;
+        }
+
+        Ok(())
+    }
+}
+
+impl Drop for NativeModuleManager {
+    /// Drop the native module manager.
+    fn drop(&mut self) {
+        //let _ = self.teardown_fs();
```

Comment:
I could enable `teardown_fs()` back, though it currently doesn't add any security, since native modules and program have the same permissions. It will be more useful when native modules only have access to a subset of the program's VFS